_source/_posts/2023-07-27-spring-graphql-react.adoc

```diff
@@ -0,0 +1,1857 @@
+  {
+    field: "companyNumber",
+    headerName: "Company #",
+    width: 100,
+    sortable: false,
+  },
+  { field: "name", headerName: "Company Name", width: 250, sortable: false },
+  { field: "category", headerName: "Category", width: 200, sortable: false },
+  { field: "SIC", headerName: "SIC", width: 200, sortable: false },
+  { field: "status", headerName: "Status", width: 100, sortable: false },
+  { field: "owner", headerName: "Owner", width: 200, sortable: false },
+];
+
+const CompanyTableContainer = (props: CompanyTableProperties) => {
+  const router = useRouter();
+  const searchParams = useSearchParams()!;
+  const pathName = usePathname();
+  const page = props.page ? props.page : 1;
+
+  const {
+    data: dataList,
+    loading: loadingList,
+    error: errorList,
+  } = useAsyncWithToken(
+    () => CompanyApi.getCompanyList({ page: page - 1}),
+    [props.page]
+  );
+
+  const { data: dataCount } = useAsyncWithToken(
+    () => CompanyApi.getCompanyCount(),
+    []
+  );
+
+  const onPageChange = (pagination: GridPaginationModel) => {
+    const params = new URLSearchParams(searchParams.toString());
+    const page = pagination.page + 1;
+    params.set("page", page.toString());
+    router.push(pathName + "?" + params.toString());
+  };
+
+
+  const companyData = dataList?.map((company: CompanyDTO) => {
+    return {
+      id: company.id,
+      name: company.name,
+      category: company.category,
+      companyNumber: company.companyNumber,
+      SIC: company.SIC,
+      status: company.status,
+      owner: company.controlledBy.map((person) => person.name).join(", "),
+    }
+  });
+
+  return (
+    <>
+      {loadingList && <Loader />}
+      {errorList && <div>Error</div>}
+
+      {!loadingList && dataList && (
+        <CompanyTable
+          pagination={{ page: page - 1, pageSize: 10 }}
+          rowCount={dataCount}
+          rows={companyData}
+          columns={columns}
+          onPageChange={onPageChange}
+        ></CompanyTable>
+      )}
+    </>
+  );
+};
+
+export default CompanyTableContainer;
+{% endraw %}
+----
+
+Give it a try, and notice how simple was the update.
+
+== Learn More
+
+I hope you enjoyed this tutorial, and found this example useful. As you can see, not much work would be required to consume more company data from the GraphQL server, just a query update in the client. Also, the Auth0 Universal Login and Auth0 React SDK provide an efficient way to secure your React applications, following security best practices. Check the Auth0 documentation for adding https://developer.auth0.com/resources/guides/spa/react/basic-authentication#add-user-sign-up-to-react[sign-up] and https://developer.auth0.com/resources/guides/spa/react/basic-authentication#add-user-logout-to-react[logout] functionality to your React application. You can find all the code for this example in the https://github.com/indiepopart/spring-graphql-react[Github repository].
```

Comment:
It looks like this needs to be broken into paragraphs.

Reply:
Added a break